srcs/http/response/http_handle_method.cpp

```diff
@@ -0,0 +1,168 @@
+	std::string       &response_body_message
+) {
+	try {
+		Stat info(path);
+		if (info.IsDirectory()) {
+			response_body_message = CreateDefaultBodyMessageFormat(
+				utils::ToString(http::FORBIDDEN), reason_phrase.at(http::FORBIDDEN)
+			);
+		} else if (info.IsRegularFile()) {
+			response_body_message = CreateDefaultBodyMessageFormat(
+				utils::ToString(http::NO_CONTENT), reason_phrase.at(http::NO_CONTENT)
+			);
+		} else {
+			// Location header fields: URI-reference
+			// ex) POST /save/test.txt HTTP/1.1
+			// Location: /save/test.txt;
+			FileCreationHandler(path, request_body_message, response_body_message);
+		}
+	} catch (const utils::SystemException &e) {
+		int error_number = e.GetErrorNumber();
+		if (error_number == ENOENT) {
+			FileCreationHandler(path, request_body_message, response_body_message);
+		} else {
+			SystemExceptionHandler(e, response_body_message);
+		}
+	}
+}
+
+void HttpResponse::DeleteHandler(const std::string &path, std::string &response_body_message) {
+	try {
+		Stat info(path);
+		if (info.IsDirectory()) {
+			response_body_message = CreateDefaultBodyMessageFormat(
+				utils::ToString(http::FORBIDDEN), reason_phrase.at(http::FORBIDDEN)
+			);
+		} else if (std::remove(path.c_str()) == 0) {
+			response_body_message = CreateDefaultBodyMessageFormat(
+				utils::ToString(http::NO_CONTENT), reason_phrase.at(http::NO_CONTENT)
+			);
+		} else {
+			throw utils::SystemException(std::strerror(errno), errno);
+		}
+	} catch (const utils::SystemException &e) {
+		SystemExceptionHandler(e, response_body_message);
+	}
+}
+
+void HttpResponse::SystemExceptionHandler(
+	const utils::SystemException &e, std::string &response_body_message
+) {
+	int error_number = e.GetErrorNumber();
+	if (error_number == EACCES || error_number == EPERM) {
+		response_body_message = CreateDefaultBodyMessageFormat(
+			utils::ToString(http::FORBIDDEN), reason_phrase.at(http::FORBIDDEN)
+		);
+	} else if (error_number == ENOENT || error_number == ENOTDIR || error_number == ELOOP || error_number == ENAMETOOLONG) {
+		response_body_message = CreateDefaultBodyMessageFormat(
+			utils::ToString(http::NOT_FOUND), reason_phrase.at(http::NOT_FOUND)
+		);
+	} else {
+		response_body_message = CreateDefaultBodyMessageFormat(
+			utils::ToString(http::INTERNAL_SERVER_ERROR),
+			reason_phrase.at(http::INTERNAL_SERVER_ERROR)
+		);
+	}
+}
+
+void HttpResponse::FileCreationHandler(
+	const std::string &path,
+	const std::string &request_body_message,
+	std::string       &response_body_message
+) {
+	std::ofstream file(path.c_str(), std::ios::binary);
+	if (file.fail()) {
+		response_body_message = CreateDefaultBodyMessageFormat(
+			utils::ToString(http::FORBIDDEN), reason_phrase.at(http::FORBIDDEN)
+		);
+		return;
+	}
+	file.write(request_body_message.c_str(), request_body_message.length());
+	file.close();
+	if (file.fail()) {
+		if (std::remove(path.c_str()) != 0) {
+			throw utils::SystemException(std::strerror(errno), errno);
+		}
```

Comment:
正常な場合は ofstream のデストラクタが close するはずなので、エラー時に明示的に close したい場合だけ close で良い気がします…！
remove の前に close する必要があるんですね、初めて知りました！
```suggestion
	file.write(request_body_message.c_str(), request_body_message.length());
	if (file.fail()) {
		file.close();
		if (std::remove(path.c_str()) != 0) {
			throw utils::SystemException(std::strerror(errno), errno);
		}
```

Reply:
ありがとうございます！書き込み以降はファイル操作しないのですぐにcloseするようにしてます！なのでcloseする必要があるかどうかはわかんないです

Reply:
remove はファイルを閉じないと処理系定義だから close 追加したのかと思いました…
https://en.cppreference.com/w/cpp/io/c/remove
> If the file is currently open by any process, the behavior of this function is implementation-defined.

逆にこういう close する必要がある時以外は自分で close しない方が良いかなと思います (多分…)